src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnectionBuilderExtensions.cs

```diff
@@ -25,12 +26,20 @@ public static IHubConnectionBuilder WithMessagePackProtocol(this IHubConnectionB
             return hubConnectionBuilder.WithHubProtocol(new MessagePackHubProtocol());
         }
 
-        public static IHubConnectionBuilder WithLogger(this IHubConnectionBuilder hubConnectionBuilder, ILoggerFactory loggerFactory)
+        public static IHubConnectionBuilder WithLoggerFactory(this IHubConnectionBuilder hubConnectionBuilder, ILoggerFactory loggerFactory)
         {
-            hubConnectionBuilder.AddSetting(HubConnectionBuilderDefaults.LoggerFactoryKey, loggerFactory);
+            hubConnectionBuilder.AddSetting(HubConnectionBuilderDefaults.LoggerFactoryKey,
+                loggerFactory ?? throw new ArgumentNullException(nameof(loggerFactory)));
```

Comment:
In general we allow passing null logger factory to HttpConnection and HubConnection so not sure why we need to throw here?

Reply:
I'm not sure it makes sense to allow a null logger factory, what would be the point of that

Reply:
OK